src/System.IO.Ports/src/System/IO/Ports/SerialPort.cs

```diff
@@ -558,43 +558,28 @@ public void DiscardOutBuffer()
         public static string[] GetPortNames()
         {
             // Hitting the registry for this isn't the only way to get the ports.
-            // 
+            //
             // WMI: https://msdn.microsoft.com/en-us/library/aa394413.aspx
             // QueryDosDevice: https://msdn.microsoft.com/en-us/library/windows/desktop/aa365461.aspx
             //
             // QueryDosDevice involves finding any ports that map to \Device\Serialx (call with null to get all, then iterate to get the actual device name)
 
-            RegistryKey baseKey = null;
-            RegistryKey serialKey = null;
-
-            String[] portNames = null;
-
-            try
+            using (RegistryKey serialKey = Registry.LocalMachine.OpenSubKey(@"HARDWARE\DEVICEMAP\SERIALCOMM"))
             {
-                baseKey = Registry.LocalMachine;
-                serialKey = baseKey.OpenSubKey(@"HARDWARE\DEVICEMAP\SERIALCOMM", false);
-
                 if (serialKey != null)
                 {
-
-                    string[] deviceNames = serialKey.GetValueNames();
-                    portNames = new String[deviceNames.Length];
-
-                    for (int i = 0; i < deviceNames.Length; i++)
-                        portNames[i] = (string)serialKey.GetValue(deviceNames[i]);
+                    string[] result = serialKey.GetValueNames();
+                    for (int i = 0; i < result.Length; i++)
+                    {
+                        // Replace the name in the array with its value.
+                        string name = result[i];
+                        result[i] = (string)serialKey.GetValue(name);
```

Comment:
I separated it out into two lines to make it more clear what was going on before I added the comment. If you prefer a single line, I can change it.